apps/admin-page/src/app/components/create-discount-dialog/create-discount-dialog.component.html

```diff
@@ -0,0 +1,86 @@
+    </div>
+    <hr />
+
+    <div class="input-container">
+        <div class="row-container">
+            <mat-form-field class="input" appearance="fill">
+                <mat-label class="input-label">Discount code</mat-label>
+                <input
+                    type="text"
+                    data-cy="discount-code-input"
+                    formControlName="discountCode"
+                    matInput
+                    placeholder="ABC123"
+                />
+                <mat-error
+                    *ngIf="
+                        createDiscountForm.get('discountCode')?.invalid && createDiscountForm.get('discountCode')?.dirty
+                    "
+                >
+                    Invalid discount - Chars: min 3, max 50
+                </mat-error>
+            </mat-form-field>
+
+            <mat-form-field class="input" appearance="fill">
+                <mat-label class="input-label">Remaining uses</mat-label>
+                <input type="number" data-cy="discount-remaining-uses-input" formControlName="remainingUses" matInput />
+            </mat-form-field>
+        </div>
+
+        <div class="row-container">
+            <mat-form-field class="input" appearance="fill">
+                <mat-label class="input-label">Starts at</mat-label>
+                <input type="date" data-cy="discount-starts-at-input" formControlName="startsAt" matInput />
+            </mat-form-field>
+
+            <mat-form-field class="input" appearance="fill">
+                <mat-label class="input-label">Expires at</mat-label>
+                <input type="date" data-cy="discount-expires-at-input" formControlName="expiresAt" matInput />
+            </mat-form-field>
+        </div>
+
+        <div class="row-container">
+            <mat-form-field class="input" appearance="fill">
+                <mat-label class="input-label">Discount type</mat-label>
+                <select matNativeControl required>
+                    <option value="PERCENT">Percent</option>
+                    <option value="AMOUNT">Amount</option>
+                </select>
+            </mat-form-field>
+
+            <mat-form-field class="input" appearance="fill">
+                <mat-label class="input-label">Amount</mat-label>
+                <input type="number" data-cy="discount-amount-input" formControlName="amount" matInput value="10" />
+            </mat-form-field>
+        </div>
+    </div>
+
+    <div class="action">
+        <button
+            mat-flat-button
+            color="accent"
+            class="action-button"
+            data-cy="discount-dialog-cancel-btn"
+            mat-dialog-close
+        >
+            Cancel
+        </button>
+        <button
+            mat-flat-button
+            color="primary"
+            class="action-button"
+            data-cy="discount-create-btn"
+            type="submit"
+            [disabled]="createDiscountForm.invalid"
+        >
+            Create
```

Comment:
(Optional) Should have a tooltip explaining why the discount is disabled
(Optional) The button should be disabled while isLoading (the request is being made)